system/install-service.sh.in

```diff
@@ -283,14 +291,20 @@ isopenrc() {
   # if there is no /etc/init.d, it's not OpenRC
   [ ! -d /etc/init.d ] && return 1
 
-  # if PID 1 is openrc-init, it's OpenRC
-  [ "$(basename "$(readlink /proc/1/exe)" 2> /dev/null)" = "openrc-init" ] && return 0
+  # if there is no rc-update command, it's not OpenRC
+  command -v rc-update > /dev/null 2>&1 || return 1
 
   # If /run/openrc/softlevel exists, it's OpenRC
   [ -f /run/openrc/softlevel ] && return 0
 
-  # if there is an openrc command, it's OpenRC
-  command -v openrc > /dev/null 2>&1 && return 0
+  # if PID 1 is openrc-init, it's OpenRC
+  [ "$(basename "$(readlink /proc/1/exe)" 2> /dev/null)" = "openrc-init" ] && return 0
```

Comment:
I think this doesn't work on my system.

I have `rc-update`, there is `/run/openrc/softlevel`, but `/proc/1/exe` links to `/sbin/init`

Reply:
I don't think that's the issue, since on Alpine `/proc/1/exe` points to `/bin/busybox`, but it detects it's OpenRC before that, because `/run/openrc/softlevel` also exists. 

@MrZammler What does `sh -x /usr/libexec/netdata/install-service.sh` show?

Reply:
This particular case is intended just to catch certain exceedingly rare situations where OpenRC is in use but does not show up according to the other checks. If booted using OpenRC, the `/run/openrc/softlevel` check should _always_ be passing, and this code should thus not be run in such cases. In cases where it’s not booted using OpenRC, the subsequent checks ending in `return 2` should be pass if OpenRC is otherwise present.

Reply:
@Dim-P :

```
6am ~ # sh -x /usr/libexec/netdata/install-service.sh
+ set -e
++ self=/usr/libexec/netdata/install-service.sh
++ '[' -L /usr/libexec/netdata/install-service.sh ']'
++ cd /usr/libexec/netdata
+++ pwd -P
++ echo /usr/libexec/netdata/install-service.sh
+ SCRIPT_SOURCE=/usr/libexec/netdata/install-service.sh
+ DUMP_CMDS=0
+ ENABLE=auto
+ EXPORT_CMDS=0
+ INSTALL=1
+ LINUX_INIT_TYPES='wsl systemd openrc lsb initd runit'
++ uname -s
+ PLATFORM=Linux
+ SVC_SOURCE=/usr/lib/netdata/system
+ SVC_TYPE=detect
+ WSL_ERROR_MSG='We appear to be running in WSL and were unable to find a usable service manager. We currently support systemd, LSB init scripts, and traditional init.d style init scripts when running under WSL.'
+ main
+ trap cleanup EXIT
+ parse_args
+ '[' -n '' ']'
+ '[' /usr/lib/netdata/system = libsysdir_POST@ ']'
+ '[' '!' -d /usr/lib/netdata/system ']'
+ valid_types
+ case "${PLATFORM}" in
+ echo 'detect systemd openrc lsb initd'
+ grep -vw detect
+ case "${PLATFORM}" in
+ '[' 1 -eq 1 ']'
+ install_linux_service
++ detect_linux_svc_type
++ '[' detect = detect ']'
++ found_types=
++ for t in ${LINUX_INIT_TYPES}
++ check_wsl
++ '[' -z '' ']'
++ iswsl
++ uname -r
++ grep -q WSL
++ uname -r
++ grep -q Microsoft
++ return 1
+ t=
+ cleanup
+ ec=1
+ '[' -n '' ']'
+ trap - EXIT
+ exit 1
```

Oks, so there must be some other reason for not installing the system service...

Reply:
Yep, it’s a bug in the auto-detection code. Should have a fix shortly.

Reply:
@MrZammler Could you remove or comment out the line reading `set -e` near the top of the script and try running it again and see if that works?

Reply:
> @MrZammler Could you remove or comment out the line reading `set -e` near the top of the script and try running it again and see if that works?

Yes, it seems to do something, but take a look at this:

```
 --- Install netdata at system init --- 
Installing OpenRC service file.
Enabling Netdata service.
 * rc-update: netdata already installed in runlevel `default'; skipping
 --- Restarting netdata instance --- 

Stopping all netdata threads
[/home/evas/stuff/netdata/netdata]# stop_all_netdata 
netdata                   | * WARNING: netdata is already stopped
 OK  ''

Starting netdata using command 'rc-service netdata start'
[/home/evas/stuff/netdata/netdata]# rc-service netdata start 
netdata                   | * Caching service dependencies ...                                                                           [ ok ]
netdata                   | * /home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/cache/netdata: correcting owner
netdata                   | * /home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/run/netdata: creating directory
netdata                   | * /home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/run/netdata: correcting owner
netdata                   | * Starting netdata ...
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0205@claim/claim.c       :load_cloud_conf): CONFIG: cannot load cloud config '/home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/lib/netdata/cloud.d/cloud.conf'. Running with internal defaults.
netdata                   |2022-10-13 16:18:24: netdata ERROR : MAIN : (0526@libnetdata/log/log.c:open_log_file  ): Cannot open file '/home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/log/netdata/debug.log'. Leaving 1 to its default. (errno 13, Permission denied)
netdata                   |2022-10-13 16:18:24: netdata ERROR : MAIN : (0526@libnetdata/log/log.c:open_log_file  ): Cannot open file '/home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/log/netdata/error.log'. Leaving 2 to its default. (errno 13, Permission denied)
netdata                   |2022-10-13 16:18:24: netdata ERROR : MAIN : (0526@libnetdata/log/log.c:open_log_file  ): Cannot open file '/home/evas/stuff/netdata/install_netdata_health_threads/netdata/var/log/netdata/access.log'. Leaving -1 to its default. (errno 13, Permission denied)
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0710@daemon/analytics.c  :get_system_time): TIMEZONE: using the contents of /etc/timezone
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0769@daemon/analytics.c  :get_system_time): TIMEZONE: fixed as 'Europe/Athens'
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0088@daemon/signals.c    :signals_init   ): SIGNAL: Not enabling reaper
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (1458@daemon/main.c       :main           ): resources control: allowed file descriptors: soft = 1024, max = 4096
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0224@daemon/daemon.c     :oom_score_adj  ): Out-Of-Memory (OOM) score is already set to the wanted value 0
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0407@daemon/daemon.c     :sched_setschedu): Adjusted netdata scheduling policy to batch (3), with priority 0.
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0332@daemon/daemon.c     :sched_getschedu): Running with process scheduling policy 'batch', nice level 19
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (1464@daemon/main.c       :main           ): netdata started on pid 2709.
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0238@spawn/spawn.c       :spawn_init     ): Initializing spawn c [ ok ]
netdata                   |2022-10-13 16:18:24: netdata INFO  : MAIN : (0801@daemon/main.c       :get_system_info): Executing /home/evas/stuff/netdata/install_netdata_health_threads/netdata/usr/libexec/netdata/plugins.d/system-info.sh
 OK  ''

Ooops! it seems netdata is not started.
Attempting another netdata start using command 'rc-service netdata start'
[/home/evas/stuff/netdata/netdata]# rc-service netdata start 
netdata                   | * WARNING: netdata has already been started
 OK  ''

Hm... it seems netdata is still not started.
 WARNING  Netdata service still not started, attempting another forced restart by running '/usr/sbin/netdata '

[/home/evas/stuff/netdata/netdata]# stop_all_netdata 
netdata                   | * Stopping netdata ...
netdata                   | * start-stop-daemon: no matching processes found                                                             [ ok ]
 OK  ''
```

Reply:
Hmm. That looks like it’s probably a bug in the installer, not the script here. We need to improve handling of detecting a running instance of Netdata (poking around and checking PIDs is _not_ the correct way to do this, we should be asking the service manager if it’s smart enough to be able to tell us, and falling back to trying to query the agent via either `netdatacli` or hitting the REST API).

Can you try manually running `rc-service netdata start` on that system (after ensuring that Netdata is _not_ running) and confirm whether the agent is running or not afterwards and whether the line for the `netdata` service in the output of `rc-status -a` says it’s running or not?

Reply:
Hi Austin.

So, I used kickstart and uninstalled netdata from the previous attempts.

Then cloned just your branch on a separate directory.

With the `set -e` present, I still got the error of installing the service file. Removing it seems to fix it, and netdata service is installed with proper paths, and starts/stops as expected.

Sorry for the previous trouble.

Reply:
@MrZammler Can you re-test with the latest commits? It should work correctly now.